src/blackdoor/cqbe/storage/StorageController.java

```diff
@@ -104,12 +104,15 @@ public Address getReferenceAddress(){
 		return isAutoRemappingEnabled() ? addressTable.getReferenceAddress() : reference;
 	}
 
-	public Address getLowest(){
+	private Address getLowest(){
 		return isAutoRemappingEnabled() && !addressTable.isEmpty() ? addressTable.firstEntry().getValue() : lowest;
 	}
 
-	public Address getHighest(){
-		return isAutoRemappingEnabled() && !addressTable.isEmpty() ? addressTable.lastEntry().getValue() : highest;
+	private Address getHighest(){
+		if(this.addressTable.size() < AddressTable.DEFAULT_MAX_SIZE)
```

Comment:
Can we add a little leeway here? So say maybe if the table is less than 90% full rather than totally full?